lib/data/dpa/dpClient.js

```diff
@@ -172,39 +230,39 @@ class DpClient {
         let fragCursor = 0;
         let dataLen = 0;
 
-        // stream a buffer
-        function streamFrag(frag, fragIdx, cb) {
+        // store a fragment
+        const streamFrag = (frag, fragIdx, cb) => {
```

Comment:
Careful with functions created inside the scope of another function. It's an optimisation killer, because it means creating the `Function` Object on every call, instead of having it available in the global context at all times.